qutebrowser/mainwindow/tabwidget.py

```diff
@@ -490,43 +506,17 @@ def tabSizeHint(self, index):
             # want to ensure it's valid in this special case.
             return QSize()
         else:
-            try:
-                pinned = self.tab_data(index, 'pinned')
-            except KeyError:
-                pinned = False
-
-            no_pinned_count = self.count() - self.pinned_count
-            pinned_width = config.val.tabs.width.pinned * self.pinned_count
+            pinned = self._tabPinned(index)
+            no_pinned_count = self.count() - self._pinnedCount()
+            pinned_width = self._tabTotalWidthPinned()
             no_pinned_width = self.width() - pinned_width
 
             if pinned:
-                size = QSize(config.val.tabs.width.pinned, height)
-                qtutils.ensure_valid(size)
-                return size
-
-            # If we *do* have enough space, tabs should occupy the whole window
-            # width. If there are pinned tabs their size will be subtracted
-            # from the total window width.
-            # During shutdown the self.count goes down,
-            # but the self.pinned_count not - this generates some odd behavior.
-            # To avoid this we compare self.count against self.pinned_count.
-            if self.pinned_count > 0 and self.count() > self.pinned_count:
-                pinned_width = config.val.tabs.width.pinned * self.pinned_count
```

Comment:
This should be properly removed from the config, which also means #2772 at least partly is a dependency of this PR. I'll try to get to it tomorrow.

Reply:
Let me know if I can add anything to do this, right now I left it out because I'm not sure how to properly remove it.

Reply:
With 7cad8f41f2d80806f15f4e3983996fcf5324ebcd you should be able to remove the option from `configdata.yml` and then run `tox -e docs`.